homeassistant/components/maico/connection.py

```diff
@@ -0,0 +1,31 @@
+"""Connection manaement."""
+import logging
+
+import httpx
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Connection:
+    """Class to manage Connection."""
+
+    def __init__(self, websession: httpx.AsyncClient, host):
+        """Initialize Connection Object."""
+        self._host = host
+        self._websession = websession
+
+    async def get(self, url, username=None, password=None):
+        """Do http get request to url."""
+        LOGGER.debug("username: %s", username)
+        if username is not None:
+            auth = httpx.BasicAuth(username, password)
```

Comment:
Is there a case where the username/password pair is expected to be none for this integration?